Lib/unittest/test/testmock/testmock.py

```diff
@@ -715,53 +715,6 @@ def method(self): pass
         self.assertRaises(StopIteration, mock.method)
 
 
-    def test_magic_method_wraps_dict(self):
-        data = {'foo': 'bar'}
-
-        wrapped_dict = MagicMock(wraps=data)
-        self.assertEqual(wrapped_dict.get('foo'), 'bar')
-        self.assertEqual(wrapped_dict['foo'], 'bar')
-        self.assertTrue('foo' in wrapped_dict)
-
-        # return_value is non-sentinel and takes precedence over wrapped value.
-        wrapped_dict.get.return_value = 'return_value'
-        self.assertEqual(wrapped_dict.get('foo'), 'return_value')
-
-        # return_value is sentinel and hence wrapped value is returned.
-        wrapped_dict.get.return_value = sentinel.DEFAULT
-        self.assertEqual(wrapped_dict.get('foo'), 'bar')
-
-        self.assertEqual(wrapped_dict.get('baz'), None)
-        with self.assertRaises(KeyError):
-            wrapped_dict['baz']
-        self.assertFalse('bar' in wrapped_dict)
-
-        data['baz'] = 'spam'
-        self.assertEqual(wrapped_dict.get('baz'), 'spam')
-        self.assertEqual(wrapped_dict['baz'], 'spam')
-        self.assertTrue('baz' in wrapped_dict)
-
-        del data['baz']
-        self.assertEqual(wrapped_dict.get('baz'), None)
-
-
-    def test_magic_method_wraps_class(self):
-
-        class Foo:
-
-            def __getitem__(self, index):
-                return index
-
-            def __custom_method__(self):
-                return "foo"
-
-
-        klass = MagicMock(wraps=Foo)
-        obj = klass()
-        self.assertEqual(obj.__getitem__(2), 2)
-        self.assertEqual(obj.__custom_method__(), "foo")
-
```

Comment:
How about leaving these tests in, but change their expectations and add comments as to why things are the way they are and how they would be in a perfect world?

Reply:
Usually reversions do a git revert of commit fixing conflicts and add a NEWS entry as needed if the change was released. I guess these can be added back as part of a separate tests only PR. I am new to doing a reversion so feel free to correct me if I am wrong.

Reply:
If cpython hadn't chose squashing commits as their appoach, I'd suggest doing the revert in one commit and then add the tests back in another commit.

However, given that squashing is the choice cpython has made, it makes little difference.
Maybe two commits in this PR which will get squashed down to one on merge?

Reply:
Okay, I have added back the test and added some comments referencing the issue.